service/controller/v19/key/key.go

```diff
@@ -438,6 +440,59 @@ func SmallCloudConfigS3URL(customObject v1alpha1.AWSConfig, accountID string, ro
 	return fmt.Sprintf("s3://%s", SmallCloudConfigPath(customObject, accountID, role))
 }
 
+// SubnetID generates Crockford Base32 (http://www.crockford.com/wrmg/base32.html)
+// encoded representation of subnet CIDR to be used as a shared identifier for
+// specific network e.g. in CloudFormation template resource names.
+func SubnetID(cidr string) (string, error) {
```

Comment:
This returned error makes me unhappy. Maybe it's an idea to use bytes of the string instead? Then we'd have a generalized method to produce IDs from arbitrary strings. Other option is just to pass *IPNet.

Reply:
I'd say that String -> String would probably expand too much (because depending on whether it's IPv4 or IPv6, one byte already takes 3-4 in string). Taking `*net.IPNet` works fine. I was considering about this during implementation but then thought that it would be painful to parse network everywhere separately instead of encapsulating it to this function.

Tim also commented to me that this could be just using some string character replacing & templating to avoid unnecessary complexity. I'll work with these comments now.

Reply:
How about converting string bytes to base58? https://en.wikipedia.org/wiki/Base58

Reply:
There is even ready golang package https://godoc.org/github.com/btcsuite/btcutil/base58

Reply:
I considered `base58` but this made me turn towards base32:
> In contrast to Base64, the digits of the encoding do not line up well with byte boundaries of the original data. For this reason, the method is well-suited to encode large integers, but not designed to encode longer portions of binary data.